primitives/src/parachain.rs

```diff
@@ -309,6 +401,40 @@ pub struct ConsolidatedIngress(pub Vec<(Id, Vec<Message>)>);
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
 pub struct BlockData(#[cfg_attr(feature = "std", serde(with="bytes"))] pub Vec<u8>);
 
+/// A chunk of erasure-encoded block data.
+#[derive(PartialEq, Eq, Clone, Encode, Decode, Default)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+pub struct ErasureChunk {
```

Comment:
Ok, we receive `chunk`, `proof` and `index` from the network, how do we know which candidate this chunk belongs to?

Reply:
The index should also be stored since availability store tries to perform block reconstruction from chunks and the index would be needed at that point

Reply:
OK, fair point. Yeah, the network packet that's gossipped should also probably contain the relay-chain block hash and the parachain ID or candidate receipt hash. I still think that should be defined specifically at the gossip level, as the `ErasureChunkMessage`.